src/group_theory/solvable.lean

```diff
@@ -175,4 +227,39 @@ begin
   ... = ⊥ : rfl,
 end
 
+lemma is_solvable_of_top_eq_bot (h : (⊤ : subgroup G) = ⊥) : is_solvable G :=
+⟨⟨0, by simp *⟩⟩
+
+variables {G} {G' : Type*} [group G'] {f : G →* G'}
+
+lemma solvable_of_solvable_injective (hf : function.injective f) (h : is_solvable G') :
```

Comment:
```suggestion
lemma solvable_of_solvable_injective (hf : function.injective f) [h : is_solvable G'] :
```

Reply:
Also elsewhere.

Reply:
Okay, I think I've replaced all of them and changed all the `lemma`s to `instance`s.